pywbem/_logging.py

```diff
@@ -0,0 +1,407 @@
+      this class and also in the python logging class from an input string
+      with the format defined in the create_loggers method below.
+
+      PywbemLoggers.create_logger - Creates a single logger fromthe
+      separate pywbem logging attributes.
+
+    """
+    __metaclass__ = MetaPywbemLoggers
+
+    @classmethod
+    def reset(cls):
+        """Reset the logger dictionary. Used primarily in unittests"""
+        cls.loggers = {}
+
+    @classmethod
+    def create_loggers(cls, input_str, log_filename=None):
+        """
+        Create the pywbem loggers defined by the input string in the following
+        syntax and place the logger definitions in the class level dictionary
+        in this class.
+
+        Parameters:
+          input_str (:term:`string`) that specifies the logger definitions:
+            as follows:
+
+            ``log_specs := log_spec[,log_spec]``
+
+            ``log_spec := log_comp['=' [dest][":"[detail_lvl][":"[log_lvl]]]]``
+
+            where:
+                ``log_comp`` must be one of strings in the LOG_COMPONENTS list.
+                ``log_lvl`` must be one of strings in the LOG_LEVELS list.
+                ``detail_lvl`` must be one of strings in the
+                    LOG_DETAIL_LEVELS list
+
+                ``dest`` must be one of strings in the LOG_DESTINATIONS list
+
+          log_filename (:term:`string`)
+            Optional string that defines the filename for output of logs
+            if the dest type is `file`
+
+        Exceptions:
+          ValueError - Generated if the syntax of the input string is invalid
+          or any of the components is not one of allowed strings.
+
+        Examples:
+            ``ops=stderr:min,debug``    # set cim operations logger
+            ``http=file::debug``        # set http logger to send to file
+            ``all=file:all:debug``      # Set all named loggers to
+        """
+        results = cls._parse_log_specs(input_str)
+        for name, value in six.iteritems(results):
+            cls.create_logger(name, log_dest=value[0],
+                              log_detail_level=value[1],
+                              log_level=value[2],
+                              log_filename=log_filename)
+
+    @classmethod
+    def create_logger(cls, log_component, log_dest='stderr',
+                      log_filename=None,
+                      log_detail_level='all', log_level='debug'):
+        """
+        Create the logger defined by the input parameters and place the result
+        in a class level dictionary in this class.
+
+        This function can be used to set up all of the named loggers used by
+        pywbem.
+
+        Parameters:
+          log_component (:term:`string`):
+           The name of the logger. It must be one of the
+           names defined in LOG_COMPONENTS. Used to create the named logger by
+           prepending with the roon logger name 'pywbem'.
```

Comment:
1. Change "named logger" to "logger name".
2. Typo "roon". Beyond that, the term "root logger" in python is used for the logger named "". that is not the case here. So I suggest "with the roon logger name 'pywbem'." -> "with the prefix `pywbem.`."